### ISSUE TITLE
Function addTileOverlay(options: TileOverlayOptions) doesn't work with tile provider

### ISSUE THREAD

**Issue by minkiapps** [bluesource/ChoiceSDK#14]
Version Choicesdk-maps 0.2

I realise that the method does not work
`addTileOverlay(TileOverlayOptions.create().tileProvider(customProviderImpl))`

By deeper dive, I see in the function 
```
override fun tileProvider(tileProvider: TileProvider?): TileOverlayOptions {
     tileProvider?.toGmsTileProvider()?.let { tileOverlayOptions.tileProvider(it) }
     return this
}

fun TileProvider.toGmsTileProvider(): com.google.android.gms.maps.model.TileProvider? {
            return when (this) {
                is GmsTileProvider -> {
                    this.getTileProvider()
                }
                else -> null
            }
        }
```

that it would just check if the implementation of TileProvider is instance of GmsTileProvider or HmsTileProvider. Library user would never have these instances of at.bluesource.choicesdk.maps.common.TileProvider.

Basically what happens at the moment is that `null` set to corresponding Mapkit and an error is thrown because no tileProvider has been set.

**Comment by officebluesource** [comment]
Issue should be fixed with [version 0.3.0](https://github.com/bluesource/ChoiceSDK/releases/tag/0.3.0).

You should create a ChoiceSDK `TileProvider` by using the `TileProvider.create(getTile: (x: Int, y: Int, zoom: Int) -> Tile?): TileProvider` function. Please also refer to the [maps documentation](https://github.com/bluesource/ChoiceSDK/blob/master/documentation/map.md).
